samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java

```diff
@@ -0,0 +1,319 @@
+import org.apache.commons.lang.Validate;
+import org.apache.samza.SamzaException;
+import org.apache.samza.sql.interfaces.Factory;
+import org.apache.samza.sql.interfaces.RelSchemaConverter;
+import org.apache.samza.sql.interfaces.RelToSamzaMsgConverter;
+import org.apache.samza.sql.interfaces.SamzaToRelMsgConverter;
+import org.apache.samza.sql.interfaces.SourceResolver;
+import org.apache.samza.sql.interfaces.SourceSchemaResolver;
+import org.apache.samza.sql.interfaces.SystemStreamConfig;
+import org.apache.samza.sql.interfaces.UdfMetadata;
+import org.apache.samza.sql.interfaces.UdfResolver;
+import org.apache.samza.sql.testutil.JsonUtil;
+import org.apache.samza.sql.testutil.ReflectionUtils;
+import org.apache.samza.sql.testutil.SamzaSqlQueryParser;
+import org.apache.samza.sql.testutil.SamzaSqlQueryParser.QueryInfo;
+import org.apache.samza.sql.testutil.SqlFileParser;
+import org.codehaus.jackson.type.TypeReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class SamzaSqlApplicationConfig {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaSqlApplicationConfig.class);
+  public static final String CFG_SQL_STMT = "samza.sql.stmt";
+  public static final String CFG_SQL_STMTS_JSON = "samza.sql.stmts.json";
+  public static final String CFG_SQL_FILE = "samza.sql.sqlFile";
+
+  public static final String CFG_FACTORY = "factory";
+
+  public static final String CFG_REL_SCHEMA_CONVERTERS = "samza.sql.relSchemaConverters";
+  public static final String CFG_FMT_REL_SCHEMA_CONVERTER_DOMAIN = "samza.sql.relSchemaConverter.%s.";
+
+  public static final String CFG_SAMZA_TO_REL_MSG_CONVERTERS = "samza.sql.samzaToRelMsgConverters";
+  public static final String CFG_FMT_SAMZA_TO_REL_MSG_CONVERTER_DOMAIN = "samza.sql.samzaToRelMsgConverter.%s.";
+
+  public static final String CFG_REL_TO_SAMZA_MSG_CONVERTERS = "samza.sql.relToSamzaMsgConverters";
+  public static final String CFG_FMT_REL_TO_SAMZA_MSG_CONVERTER_DOMAIN = "samza.sql.relToSamzaMsgConverter.%s.";
+
+  public static final String CFG_SCHEMA_RESOLVERS = "samza.sql.schemaResolvers";
+  public static final String CFG_FMT_SCHEMA_RESOLVER_DOMAIN = "samza.sql.schemaResolver.%s.";
+
+  public static final String CFG_SOURCE_RESOLVER = "samza.sql.sourceResolver";
+  public static final String CFG_FMT_SOURCE_RESOLVER_DOMAIN = "samza.sql.sourceResolver.%s.";
+
+  public static final String CFG_UDF_RESOLVER = "samza.sql.udfResolver";
+  public static final String CFG_FMT_UDF_RESOLVER_DOMAIN = "samza.sql.udfResolver.%s.";
+
+//  private final String _selectQuery;
+
+  private SourceResolver sourceResolver;
+  private UdfResolver udfResolver;
+
+  private final Collection<UdfMetadata> udfMetadata;
+
+  private final Map<String, SystemStreamConfig> inputSystemStreamConfigBySource;
+
+  private final Map<String, SystemStreamConfig> outputSystemStreamConfigsBySource;
+
+  private Map<String, SamzaToRelMsgConverter> samzaToRelMsgConvertersByName = new HashMap<>();
```

Comment:
It looks like at least some of these could be final. Also if you want to maintain an invariant that they are immutable you can use an immutable wrapper in the constructor.